core/src/components/datetime/datetime.ios.scss

```diff
@@ -92,14 +92,15 @@
   font-size: 20px;
 }
 
-:host .calendar-day:after {
+.calendar-day:focus .calendar-day-highlight,
+.calendar-day.calendar-day-active .calendar-day-highlight {
+  // !important is needed to overwrite custom highlight background, which is inline.
+  /* stylelint-disable-next-line declaration-no-important */
+  background: current-color(base) !important;
```

Comment:
Can you explain why `!important` is needed here?

`current-color(base)` evaluates to: `var(--ion-color-base)`, which has a value of: `var(--ion-color-primary, #3880ff)!important`.

So this style is evaluating as:

```css
background: var(--ion-color-primary, #3880ff)!important !important;
```

I removed the `!important` from your stylesheet and tested against iOS mode, but do not see a visual difference.